WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -136,14 +138,32 @@ private extension PointOfSaleDashboardViewModel {
 
     func observeCardPresentPaymentEvents() {
         cardPresentPaymentService.paymentEventPublisher.assign(to: &$cardPresentPaymentEvent)
+        cardPresentPaymentService.paymentEventPublisher
+            .map { event in
+                guard case let .showAlert(alertDetails) = event else {
+                    return nil
+                }
+                return alertDetails
+            }
+            .assign(to: &$cardPresentPaymentAlertViewModel)
+        cardPresentPaymentService.paymentEventPublisher
+            .map { event in
+                guard case let .showPaymentMessage(messageType) = event else {
+                    return nil
+                }
+                return messageType
+            }
+            .assign(to: &$cardPresentPaymentInlineMessage)
         cardPresentPaymentService.paymentEventPublisher.map { event in
             switch event {
-            case .idle:
+            case .idle, .showPaymentMessage:
                 return false
-            case .showAlert,
-                    .showReaderList,
-                    .showOnboarding:
+            case .showAlert:
                 return true
+            case .showReaderList,
```

Comment:
I think we should flatten `showReaderList` into `showAlert`, because we can do that now that we have control over the view models emitted, and it's just another part of the connection flow.